src/model-manager.ts

```diff
@@ -1,33 +1,29 @@
-const Toposort = require('toposort-class');
-const _ = require('lodash');
+import Toposort from 'toposort-class';
+import * as _ from 'lodash';
+import Model from './model';
+import { TableName } from './common';
 
-class ModelManager {
-  constructor(sequelize) {
-    this.models = [];
-    this.sequelize = sequelize;
-  }
+export class ModelManager {
+  private models: typeof Model[] = [];
+  constructor(private sequelize: any) {}
 
-  addModel(model) {
+  public addModel(model: typeof Model): typeof Model {
```

Comment:
Should probably be generic.

Reply:
We don't have generic models yet so I figured we'd revisit each use of Model anyways. Do you want me to change it to: ?

```suggestion
  public addModel<T extends typeof Model>(model: T): T {
```

Reply:
Basically yes.